cloudflare_dns: some refactoring

russoz opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
The purpose of the removed validation `if` block is defeated by the `required_if` clause in line 949.

Either the `if` block was wrong and should be removed - in which case this is a refactoring PR, or the `required_if` was wrong, then the `if` block should be restored and the `required_if` clause should be removed, in which case this is a bug fix PR.

There are no tests for this module. 

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

<!--- Please do not forget to include a changelog fragment:
      https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-changelog-fragments
      No need to include one for docs-only or test-only PR, and for new plugin/module PRs.
      Read about more details in CONTRIBUTING.md.
      -->

##### ISSUE TYPE
<!--- Pick one or more below and delete the rest.
      'Test Pull Request' is for PRs that add/extend tests without code changes. -->
- Bugfix Pull Request
- Refactoring Pull Request

##### COMPONENT NAME
<!--- Write the SHORT NAME of the module, plugin, task or feature below. -->
cloudflare_dns

felixfontein commented:
Ref: https://github.com/ansible-collections/community.general/pull/7399#discussion_r1364396419

russoz commented:
I was re-reading the changes and I noticed that the validation code I removed in the first commit was not checking just `tag` and `flag`, but also `value`. So I reverted that for now.

felixfontein reviewed (approved):
This is quite a large change, quite also quite necessary :) Thanks for doing this!
